library/prolog_colour.pl

```diff
@@ -1605,6 +1605,70 @@
 colourise_list_args([], TP, T, TB, How) :-
     specified_item(How, T, TB, TP).
 
+
+%!  colourise_expression(+Term, +TB, +Pos)
+%
+%   colourise arithmetic expressions.
+
+colourise_expression(_, _, Pos) :-
+    var(Pos),
+    !.
+colourise_expression(Arg, TB, parentheses_term_position(PO,PC,Pos)) :-
+    !,
+    colour_item(parentheses, TB, PO-PC),
+    colourise_expression(Arg, TB, Pos).
+colourise_expression(Var, TB, Pos) :-                     % variable
+    var(Var), Pos = _-_,
```

Comment:
You can have rules for all the stuff you want to take care of and call colourise_term_arg/3 to do the general coloring.    That avoids some duplication.   That applies to variables and numeric types, I guess.   Note that you can have `Dict.key`, so you should also propagate ./2 terms.

Reply:
Thanks, done in [724822e](https://github.com/SWI-Prolog/swipl-devel/pull/1101/commits/724822ee2757354a926d45c8ab0c99fb1feb5165).  I've added a simple test case that covers also `Dict.key` in expressions in another commit.